segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -177,6 +180,29 @@ private MetadataStore createMetadataStore() {
         return builder.build();
     }
 
+    /**
+     * Initializes storage.
+     *
+     * @throws Exception
+     */
+    private void initializeStorage() throws Exception {
+        this.storage.initialize(this.metadata.getContainerEpoch());
+
+        if (this.storage instanceof ChunkedSegmentStorage) {
```

Comment:
This instanceof someone smells.
Btw you have to cast it to ChunkedSegmentStorage so I don't know if we can do it better

Reply:
I think this will go away once we remove the legacy code (RollingStorage and AsyncStorageWrapper). Until this code comes out of beta, we'll have to live with both.

Reply:
okay